src/util/arith_tools.cpp

```diff
@@ -299,12 +323,36 @@ bool get_bvrep_bit(
 irep_idt
 make_bvrep(const std::size_t width, const std::function<bool(std::size_t)> f)
 {
-  std::string result(width, ' ');
+  std::string result;
+  result.reserve(width / 4 + 1);
+  unsigned nibble = 0;
 
   for(std::size_t i = 0; i < width; i++)
-    result[width - 1 - i] = f(i) ? '1' : '0';
+  {
+    const auto bit_in_nibble = i % 4;
 
-  return result;
+    nibble |= ((unsigned)f(i)) << bit_in_nibble;
```

Comment:
What about:
```
if(f(i))
  nibble |= 1u << bit_in_nibble;
```
Not sure to what extent performance would differ given the additional branch, though.